### ISSUE TITLE
Breaks Statsample 1.4.0

### ISSUE THREAD

**Issue by einpaule** [SciRuby/statsample-bivariate-extension#3]
It seems that the latest version of this gem 1.2.0 (https://rubygems.org/gems/statsample-bivariate-extension/versions/1.2.0 ) prevents us from using statsample 1.4.0. It results in an infinite loop of dependency:

``` console
Your Gemfile requires gems that depend on each other, creating an infinite loop. Please remove either gem 'statsample' or gem 'statsample-bivariate-extension' and try again.
```

It can be reproduced by using bundler with this `Gemfile` file in an empty folder:

``` bundler
source 'https://rubygems.org'

gem 'statsample', '1.4.0'
```

I'm not sure whether this is something you would want to support, nor what the solution is.

**Comment by v0dro** [comment]
There are pretty big differences between statsample `1.4.0` and `2.0.1`. The updated bivariate-extensions uses `2.0.1` because they both now use [daru](www.github.com/v0dro/daru) for storage of data.

 Is there some specific reason why you're not updating to the latest statsample? Could you share it here? If it's critical then maybe we can think of adding support.

@agarie

**Comment by einpaule** [comment]
We've now upgraded to 1.4.3, the original issue why we stuck to 1.4.0 was:
https://github.com/SciRuby/statsample/issues/21#issuecomment-86559933

but now I'm applying the workaround I suggested to this issue (check whether all values are equal before calling statsample), so we're no longer affected by the issue and we were able to go to 1.4.3 yesterday and this solves the issue.

**TL;DR: We moved to statsample 1.4.3, this issue disappeared and you can still use it with Rails 3**

**Comment by v0dro** [comment]
Cool!

Could you send a PR to statsample with your fixes?

**Comment by einpaule** [comment]
Well, we actually implemented the check in the wrapper we have around statsample. 

The problem with implementing this in the statsample code is that since the retrieval of the values does the calculation, there's no really elegant way of doing this check. I.e. whenever I retrieve the coefficients, I would have to check whether the y_var is constant or not and return it.

I don't have a really good idea for doing this that would not break the interface.

**Comment by bitsapien** [comment]
I'm using the latest version of `statsample`, but I still get this problem -> 
```
Your bundle requires gems that depend on each other, creating an infinite loop. Please
remove either gem 'statsample' or gem 'statsample-bivariate-extension' and try again.
```
I'm using Rails 4.2. I tried locking the gem version to `2.0.1`, but its dependency `reportbuilder`'s version is conflicting with `prawn`'s version.
```
  In snapshot (Gemfile.lock):
    prawn (= 2.1.0)

  In Gemfile:
    prawn (~> 2.1.0)

    statsample (>= 2.0.1) was resolved to 2.0.1, which depends on
      reportbuilder (~> 1.4) was resolved to 1.4.0, which depends on
        prawn-svg (~> 0.9.1) was resolved to 0.9.1.11, which depends on
          prawn (>= 0.8.4)

    statsample (>= 2.0.1) was resolved to 2.0.1, which depends on
      reportbuilder (~> 1.4) was resolved to 1.4.0, which depends on
        prawn (~> 0.8.4)
```

I think downgrading my gems would help, but I dont want to do that.
Please help.
